### PR TITLE
User file with PTHP has negative heating electricity 

### PR DESCRIPTION
**Pull request by Nigusse**
Issue #6396

Addresses PTHP with negative heating electricity due to EIR modifier curve as function of temperature curve value being less than zero. This is caused by bad curve. This issue also impacts standard rating heating capacity for DX heating coils. 

Pull request overview
---------------------
This is a defect caused by bad EIR modifier curves. If any of the EIR or capacity modifier curves as a function of temperature results in negative value, then the curve value is set to zero and recurring warning error message is generated. Also severe warning error message for standard ratings calculation.
Diffs are not expected for this change.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things

### CODE REVIEW

**Comment by Nigusse** [comment]
@mjwitte I check for the value of heating capacity and EIR function of temperature curves. If any of the curves value is less than zero, I set HSPF value to zero and throw error message. Also if the capacity modifier curves value are less than zero, I set the standard ratings net heating capacity value to zero. In both cases the simulation continues. Now I see 28 example files showing diffs in EIO, err and eplustbl.htm files. These diffs are causes by new error messages and HSPF values being set to zero. I will selectively investigate the curves in some of those idf with diffs.

File: tst/EnergyPlus/unit/StandardRatings.unit.cc [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+		// evaluate EIR curves
+		Real64 EIRTempModFacRated = CurveValue( Coil.EIRFTemp( 1 ), StandardRatings::HeatingOutdoorCoilInletAirDBTempRated );
+		Real64 EIRTempModFacH2Test = CurveValue( Coil.EIRFTemp( 1 ), StandardRatings::HeatingOutdoorCoilInletAirDBTempH2Test );
+		Real64 EIRTempModFacH3Test = CurveValue( Coil.EIRFTemp( 1 ), StandardRatings::HeatingOutdoorCoilInletAirDBTempH3Test );
+		// check EIR curve value
+		EXPECT_LT( EIRTempModFacRated, 0.0 );
+		EXPECT_GT( EIRTempModFacH2Test, 0.0 );
+		EXPECT_GT( EIRTempModFacH3Test, 0.0 );
+		// if one of the CAP or EIR curves value is less than zero, then HSPF is set to zero
+		EXPECT_DOUBLE_EQ( HSPF, 0.0 );
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/StandardRatings.unit.cc:230]: :+1:

File: tst/EnergyPlus/unit/StandardRatings.unit.cc [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+		Coil.DefrostTime = 0.058333;
+		Coil.DefrostCapacity = 1000;
+		Coil.PLRImpact = false;
+		Coil.FuelType = FuelTypeElectricity;
+		Coil.RegionNum = 4;
+		Coil.OATempCompressorOn = -5.0;
+		Coil.OATempCompressorOnOffBlank = "true";
+		NumCurves = 5;
+		PerfCurve.allocate( NumCurves );
+		PerfomanceCurveData * pCurve;
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/StandardRatings.unit.cc:120]: This is OK here.  This pointer isn't being used to store anything, it's just a reference.  For "reference", you don't need to do this complexity.  Just create an actual reference:

```
auto & thisCurve = PerfCurve( nCapfT );
thisCurve.CurveType = Cubic;
```

Then you don't have the sometimes yellow/red flag of a pointer variable during review.  The only downside I can think of is that you have to create multiple variable names below, one for each curve.  Unless you put them inside individual scopes (braces).

Anyway, I see your intent here, and it's OK for here, but in the future, consider the alternative approach.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Built locally, confirmed the energy use is now zero with the curve values being constrained to 0.  I also merged develop and resolved the tiny conflict.  I'm pushing it up as soon as it builds successfully here.  If CI is happy, this will merge.

File: tst/EnergyPlus/unit/StandardRatings.unit.cc (hunk 2 above)
Comment (Nigusse) [reply tst/EnergyPlus/unit/StandardRatings.unit.cc:120]: @Myoldmopar  Thanks. Comment noted.

**Comment by Myoldmopar** [comment]
I deleted the Mac results there and we'll see if my regression fixes help get rid of those audit diffs.

**Comment by Myoldmopar** [comment]
As much as I hoped seeing the audit diffs go away, they haven't yet.  I think it might not get a new regression repo clone until it finishes its current set of known builds.  We'll see later.

**Pull request merged**